tests/python_test.py

```diff
@@ -487,8 +504,85 @@ def test_write_resource_invalid_disk(tmpdir, sanlock_daemon, disk):
         sanlock.write_resource("ls_name", "res_name", disks)
     assert repr(disk) in str(e.value)
 
+
+@contextmanager
+def raises_sanlock_errno(expected_errno=errno.ECONNREFUSED):
```

Comment:
robably we had another sanlock process still running from another crashed test.
so we had our foreground test communicating with the still active process and getting the ENOENT replies.
this can be easily reproduced:
1. run tox in one terminal will all tests
2. run tox -- -k parse_args in another terminal for the relevant tests
and you'll get the different errnos by mistake.